src/main/java/meetup/connect/googleApi/GoogleCalendar.java

```diff
@@ -0,0 +1,188 @@
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Service;
+import org.springframework.util.CollectionUtils;
+
+import java.io.*;
+import java.security.GeneralSecurityException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.function.Supplier;
+
+
+@Service
+public class GoogleCalendar {
+  private static final String APPLICATION_NAME = "MeetUpConnect Calendar";
+
+  /** Global instance of the JSON factory. */
+  private static final JsonFactory JSON_FACTORY = GsonFactory.getDefaultInstance();
+
+  /** Directory to store authorization tokens for this application. */
+  private static final String TOKENS_DIRECTORY_PATH = "token";
+  private static final String CIRCUIT_BREAKER_NAME = "googleApiService";
+  private static final Integer PORT = 8888;
+
+  private static final String CALENDAR_ID = "primary";
+  private static final Logger log = LoggerFactory.getLogger(GoogleCalendar.class);
+
+  /**
+   * Global instance of the scopes required by this quickstart. If modifying these scopes, delete
+   * your previously saved tokens/ folder.
+   */
+  private static final List<String> SCOPES = List.of(CalendarScopes.CALENDAR_EVENTS);
+
+  private final CalendarProperties calendarProperties;
+  private static Calendar service;
+
+  public GoogleCalendar(CalendarProperties calendarProperties) {
+    this.calendarProperties = calendarProperties;
+  }
+
+  private Credential getCredentials() throws IOException {
+    GoogleAuthorizationCodeFlow flow =
+        new GoogleAuthorizationCodeFlow.Builder(
+                new NetHttpTransport(),
+                GsonFactory.getDefaultInstance(),
+                calendarProperties.getClient_id(),
+                calendarProperties.getClient_secret(),
+                SCOPES)
+            .setDataStoreFactory(new FileDataStoreFactory(new java.io.File(TOKENS_DIRECTORY_PATH)))
+            .setAccessType("offline")
+            .build();
+    LocalServerReceiver receiver = new LocalServerReceiver.Builder().setPort(PORT).build();
+    // returns an authorized Credential object.
+    return new AuthorizationCodeInstalledApp(flow, receiver).authorize("user");
+  }
+
+  private Calendar getService() throws GeneralSecurityException, IOException {
+    final NetHttpTransport HTTP_TRANSPORT = GoogleNetHttpTransport.newTrustedTransport();
+    var requestFactory = HTTP_TRANSPORT
+            .createRequestFactory(httpRequest -> httpRequest
+                    .setConnectTimeout(10)
+                    .setReadTimeout(5)
+                    .setNumberOfRetries(3)
+            )
+            .getTransport();
+    return new Calendar.Builder(requestFactory, JSON_FACTORY, getCredentials())
+            .setApplicationName(APPLICATION_NAME)
+            .build();
+  }
+
+  private Supplier<Calendar> getGoogleCalendarSupplier() {
+    CircuitBreakerConfig googleApiServiceConfig = CircuitBreakerConfig.ofDefaults();
```

Comment:
issue: you should be more precise with specifying when to trigger the circuit breaker

Reply:
https://resilience4j.readme.io/docs/examples#create-a-circuitbreakerregistry

Reply:
e.g.  `.recordExceptions(GeneralSecurityException.class, IOException.class)`